coder ssh nil pointer dereference under heavy load / agent disconnected

During scaletesting where the coder server was under heavy load from multiple agents and API responses very slow, we observed the following nil pointer dereference:

```console
❯ coder ssh scaletest/runner
==> ⧗ The workspace agent lost connection
2023-09-27 10:49:47.956Z Wait for it to reconnect or restart your workspace.
2023-09-27 10:49:47.956Z For more information and troubleshooting, see https://coder.com/docs/v2/latest/templates#agent-connection-issues and https://coder.com/docs/coder-oss/latest/templates#troubleshooting-templates
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12ef680]

goroutine 1 [running]:
github.com/coder/coder/v2/cli/cliui.Agent({0x1c60878?, 0xc00070d1d0?}, {0x1c51540?, 0xc000130010}, {0x62, 0xf6, 0x9c, 0x9a, 0xd8, 0xac, ...}, ...)
	/home/runner/actions-runner/_work/coder/coder/cli/cliui/agent.go:233 +0xb00
github.com/coder/coder/v2/cli.(*RootCmd).ssh.func1(0xc0003b1e00)
	/home/runner/actions-runner/_work/coder/coder/cli/ssh.go:186 +0x113e
github.com/coder/coder/v2/cli.(*RootCmd).Command.func6.1(0xc0001c36c0?)
	/home/runner/actions-runner/_work/coder/coder/cli/root.go:277 +0xd1
github.com/coder/coder/v2/cli.(*RootCmd).initClientInternal.func1.1(0xc0003b1e00)
	/home/runner/actions-runner/_work/coder/coder/cli/root.go:596 +0x882
github.com/coder/coder/v2/cli/clibase.RequireRangeArgs.func1.1(0x0?)
	/home/runner/actions-runner/_work/coder/coder/cli/clibase/cmd.go:574 +0x362
github.com/coder/coder/v2/cli/clibase.(*Invocation).run(0xc0003b1e00, 0xc0002ade00)
	/home/runner/actions-runner/_work/coder/coder/cli/clibase/cmd.go:388 +0x10b2
github.com/coder/coder/v2/cli/clibase.(*Invocation).run(0xc0003b1b00, 0xc0002ade00)
	/home/runner/actions-runner/_work/coder/coder/cli/clibase/cmd.go:325 +0xb5e
github.com/coder/coder/v2/cli/clibase.(*Invocation).Run(0xc0002adeb8?)
	/home/runner/actions-runner/_work/coder/coder/cli/clibase/cmd.go:478 +0xe5
github.com/coder/coder/v2/cli.(*RootCmd).RunMain(0xc000547e30?, {0xc00053ae00, 0x25, 0x40})
	/home/runner/actions-runner/_work/coder/coder/cli/root.go:134 +0x1a5
main.main()
	/home/runner/actions-runner/_work/coder/coder/enterprise/cmd/coder/main.go:11 +0x45
```

The coder client version used was the latest (at the time of writing): [`v2.1.5+66019ef`](https://github.com/coder/coder/commit/66019ef70a69cc65679733da65547761a67e50ce)

The nil pointer deref happens at: https://github.com/coder/coder/blob/c67db6efb089d9a9f79fe153a2f9a7b3705f949a/cli/cliui/agent.go#L245

This suggests last connected or disconnected may be nil.

--- a/cli/cliui/agent.go
+++ b/cli/cliui/agent.go
@@ -237,12 +237,14 @@ func Agent(ctx context.Context, writer io.Writer, agentID uuid.UUID, opts AgentO
 			sw.Start(stage)
 			sw.Log(time.Now(), codersdk.LogLevelWarn, "Wait for it to reconnect or restart your workspace.")
 			sw.Log(time.Now(), codersdk.LogLevelWarn, troubleshootingMessage(agent, "https://coder.com/docs/v2/latest/templates#agent-connection-issues"))
+
+			disconnectedAt := *agent.DisconnectedAt
 			for agent.Status == codersdk.WorkspaceAgentDisconnected {
 				if agent, err = fetch(); err != nil {
 					return xerrors.Errorf("fetch: %w", err)
 				}
 			}
-			sw.Complete(stage, agent.LastConnectedAt.Sub(*agent.DisconnectedAt))
+			sw.Complete(stage, agent.LastConnectedAt.Sub(disconnectedAt))
 		}
 	}
 }

--- a/cli/cliui/agent_test.go
+++ b/cli/cliui/agent_test.go
@@ -133,6 +133,7 @@ func TestAgent(t *testing.T) {
 				},
 				func(_ context.Context, agent *codersdk.WorkspaceAgent, _ chan []codersdk.WorkspaceAgentLog) error {
 					agent.Status = codersdk.WorkspaceAgentConnected
+					agent.DisconnectedAt = nil
 					agent.LastConnectedAt = ptr.Ref(time.Now())
 					return nil
 				},

